### ISSUE TITLE
TestResources tenantId requirement

### ISSUE THREAD

**Issue by JoshLove-msft** [Azure/azure-sdk-tools#861]
The [README](https://github.com/Azure/azure-sdk-for-net/blob/master/eng/common/TestResources/README.md) for TestResources has this snippet:
```powershell
Connect-AzAccount -Subscription 'YOUR SUBSCRIPTION ID'
$sp = New-AzADServicePrincipal -Role Owner
eng\common\TestResources\New-TestResources.ps1 `
  -BaseName 'myusername' `
  -ServiceDirectory 'search' `
  -TestApplicationId $sp.ApplicationId `
  -TestApplicationSecret (ConvertFrom-SecureString $sp.Secret -AsPlainText)
```
Running this snippet will not work as is since the user will be prompted to enter the `tenantId`, since the test-resources.json requires it (it appears that all test-resources.json files require tenantId even if they are not using it). If the user tries to include the tenantId as an argument, then they are prompted to include the `ProvisionerApplicationId`, due to these parameters being in the same parameter set. It doesn't seem like `tenantId` is actually needed when not specifying `ProvisionerApplicationId`. It is computed from the context for the [output](https://github.com/Azure/azure-sdk-for-net/blob/master/eng/common/TestResources/New-TestResources.ps1#L317), and it seems like this can be done as well for the value that is passed to test-resources.json. 

To sum it up - is it possible to just pass the context.Tenant.Id to the test-resources.json script so that users will not be prompted for it unless they are also including `ProvisionerApplicationId`?

**Comment by heaths** [comment]
You can pass it, sure, but if a test-resources.json doesn't need a tenantId, you can remove it from test-resources.json. You can also use a default value of our tenant since that's where it's going to be used most.

**Comment by JoshLove-msft** [comment]
> You can pass it, sure, but if a test-resources.json doesn't need a tenantId, you can remove it from test-resources.json. You can also use a default value of our tenant since that's where it's going to be used most.

The docs for test-resources.json say that all parameters should be included:
https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/48/Create-a-new-Live-Test-pipeline?anchor=test-resources.json

**Comment by heaths** [comment]
Fixed that. 🙂

**Comment by weshaggard** [comment]
The tenant id is definitely one of the things I was thinking we should default when possible with my simplification ideas in https://github.com/Azure/azure-sdk-tools/issues/541.

**Comment by JoshLove-msft** [comment]
> The tenant id is definitely one of the things I was thinking we should default when possible with my simplification ideas in #541.

Should we just use `context.Tenant.Id`?

**Comment by weshaggard** [comment]
Yes I was expecting we would default it from the current Azure Context.

**Comment by kurtzeborn** [comment]
Core problem covered by this issue: https://github.com/Azure/azure-sdk-tools/issues/541

Closing...
